src/librustdoc/html/render/mod.rs

```diff
@@ -4414,6 +4423,25 @@ fn sidebar_deref_methods(impl_: &Impl, v: &Vec<Impl>) -> String {
                 out.push_str(&format!("<div class=\"sidebar-links\">{}</div>", ret.join("")));
             }
         }
+
+        // Recurse into any further impls that might exist for `target`
+        if let Some(target_did) = target.def_id() {
+            if let Some(target_impls) = c.impls.get(&target_did) {
+                if let Some(target_deref_impl) = target_impls
+                    .iter()
+                    .filter(|i| i.inner_impl().trait_.is_some())
+                    .find(|i| i.inner_impl().trait_.def_id() == c.deref_trait_did)
+                {
+                    if let Some(type_did) = impl_.inner_impl().for_.def_id() {
+                        if target_did == type_did {
+                            // Avoid infinite cycles
+                            return out;
+                        }
+                    }
```

Comment:
Why do you need to check this both here and on line 3637 above?

Reply:
`sidebar_deref_methods` and the other sidebar code paths are called separately from rendering the main page contents via `render_deref_methods` and other render paths. At the moment, both the main body rendering and sidebar rendering have their own recursion, so that's why they both need to guard against cycles.